### PR TITLE
feat: move core rpc server types to standalone crate 

### LINKED ISSUE #8460: Extract core rpc server types to standalone crate
### Describe the feature

currently node-core pulls in rpc-builder for constants and setup code

rpc-builder is pretty beefy because it pulls in the entire stack required for all endpoints.
node-core sits pretty deep in the stack so this results in bloated dependency trees

## TODO
[x] add new crate `rpc-server(-common|types)` and extract all types/constants used by node-core rn
[ ] move eth cache types https://github.com/paradigmxyz/reth/blob/13fcb885c60bab1390c4525cb5a0c35ab215e833/crates/rpc/rpc/src/eth/cache/config.rs#L2-L2

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by leruaa**
ref #8460

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -286,6 +286,7 @@ reth-rpc-builder = { path = "crates/rpc/rpc-builder" }
 reth-rpc-engine-api = { path = "crates/rpc/rpc-engine-api" }
 reth-rpc-layer = { path = "crates/rpc/rpc-layer" }
 reth-rpc-types = { path = "crates/rpc/rpc-types" }
+reth-rpc-server-types = { path = "crates/rpc/rpc-server-types" }
```
Comment (mattsse) [inline Cargo.toml:289]: swap order

File: crates/node-core/Cargo.toml [hunk 2]
```diff
@@ -22,6 +22,7 @@ reth-network-p2p.workspace = true
 reth-rpc-builder.workspace = true
 reth-rpc.workspace = true
 reth-rpc-types.workspace = true
+reth-rpc-server-types.workspace = true
```
Comment (mattsse) [inline crates/node-core/Cargo.toml:25]: swap order, alphabetically

File: crates/rpc/rpc-builder/src/lib.rs [hunk 3]
```diff
@@ -192,6 +191,7 @@ use reth_rpc::{
 };
 use reth_rpc_api::servers::*;
 use reth_rpc_layer::{AuthLayer, Claims, JwtAuthValidator, JwtSecret};
+use reth_rpc_server_types::constants::*;
```
Comment (mattsse) [inline crates/rpc/rpc-builder/src/lib.rs:194]: this should be re-exported as `pub use reth_rpc_server_types::constants;`

File: crates/rpc/rpc-server-types/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+homepage.workspace = true
+repository.workspace = true
+description = "RPC server types and constants"
+
+[lints]
+workspace = true
+
+[dependencies]
+# reth
+reth-rpc.workspace = true
```
Comment (mattsse) [inline crates/rpc/rpc-server-types/Cargo.toml:16]: it'd be great if we can do without.

no that we have a separate crate for constants we can move all constants from reth-rpc crate in here and then import in reth-rpc instead, inverting the dependency.

**Review by mattsse — CHANGES_REQUESTED** [review]
great start

with a standalone, lightweight crate we can also pull this in in reth-rpc and move the constants to reth-rpc-server-types

File: crates/node-core/Cargo.toml [hunk 5]
```diff
@@ -21,6 +21,7 @@ reth-network = { workspace = true, features = ["serde"] }
 reth-network-p2p.workspace = true
 reth-rpc-builder.workspace = true
```
Comment (mattsse) [inline crates/node-core/Cargo.toml:22]: this we want to remove

File: crates/rpc/rpc-server-types/src/lib.rs
Comment (mattsse) [inline crates/rpc/rpc-server-types/src/lib.rs:1]: this needs the regular lib.rs header and unused clippy attribute, see other crates

File: crates/rpc/rpc-server-types/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+homepage.workspace = true
+repository.workspace = true
+description = "RPC server types and constants"
+
+[lints]
+workspace = true
+
+[dependencies]
+# reth
+reth-primitives.workspace = true
```
Comment (mattsse) [inline crates/rpc/rpc-server-types/Cargo.toml:16]: if possible depend on alloy-primitives instead

**Review by mattsse — CHANGES_REQUESTED** [review]
solid,

next we need the remaining types that are used in node-core

**Comment by leruaa** [comment]
@mattsse Can you please have a look at my last commit and tell me if I'm heading in the right direction? Feels like I'm moving the entire `rpc-builder` crate...

**Comment by mattsse** [comment]
I see, the main problem is this config trait:

https://github.com/paradigmxyz/reth/blob/e75edf3be7d2fc48e510737a06b8717d26bb09b3/crates/node-core/src/cli/config.rs#L18-L18

hmm, this is a bit tricky, I need to think about this a bit,

okay, for this pr, let's only do constants first, get the pr over the line and then continue with more incremental changes. it's fine if node-core then also depends on server-types.
but it will be easier to move stuff around once we have the crate

File: crates/node-core/Cargo.toml (hunk 5 above)
Comment (leruaa) [reply crates/node-core/Cargo.toml:22]: Aborted for now

**Comment by leruaa** [comment]
@mattsse I think this is ready to be reviewed.

P.S.: Don't hesitate if you can think of some good first issues for the next steps :)

**Review by mattsse — APPROVED** [review]
lgtm

next we can move these constants and types for cache

https://github.com/paradigmxyz/reth/blob/13fcb885c60bab1390c4525cb5a0c35ab215e833/crates/node-core/src/args/rpc_state_cache.rs#L2-L5

then I think we move the config traits

**Pull request merged**